### DISCUSSION TITLE
Can neural prophet model available for multivariates?

### DISCUSSION THREAD

**Discussion by landkwon94** [ourownstory/neural_prophet#1388] [Q&A - get help using NeuralProphet]
First of all, many thanks for the contributions!

I have a question about this model usage.

I am studying all tutorials in this github, and I see that most tutorials are about uni-variate forecasting.

As I want to perform multivariate time series forecasting, I guess these codes :

col_lst = ['precipitation', 'soil_moisture', 'solar_radiation', 'temperature', 'wind_speed']
m = m.add_lagged_regressor(names=col_lst, normalize="minmax") 

may train the model for multivariate forecasting.

May I ask am I correct?
And is there any tutorials which deal with multi-variate? I want to study if there is :) 

Another question is that, is it possible to add regional information for training the model?
For example, not only numeric values (e.g., precipitation, temperature, etc.), I want to add regions (e.g., State, Country, etc.) for training.

I guess tutorial in here : https://github.com/ourownstory/neural_prophet/blob/main/docs/source/tutorials/tutorial09.ipynb
may be related to my question. But just curious, is it possible to train the model with regional values? 

Again, super thanks for your prompt reply!!!

**Comment by leoniewgnr** [comment]
Hi @landkwon94, thanks for your question.
**Multivariate forecasting**
Yes, NP can do multivariate forecasting. This tutorial describes how to add regressors: https://neuralprophet.com/tutorials/tutorial05.html
You can just do it like the code your code snippet.

**Regional information**
That is also possible. We call it a global local model. Because some parts are modelled globally (over all regions) and some local (individual for each region). Feel free to check out this tutorial: https://neuralprophet.com/tutorials/tutorial09.html

Hope this helps! Let me know if you have further questions!

**Comment by lvzhiqiang** [comment]
Hi @leoniewgnr , thanks for your answer with multivariate forecasting. However, the turorial05 only contains 2 timeseries. How to deal with more than 3 timeseris ? I could't find any parameter in the api. Thanks a lot!

**Reply by leoniewgnr** [reply]
Just add more IDs:) It will recognize it automatically

**Comment by weidongzhou1994** [comment]
Can I use the global and multivariate for multi-goods sales prediction?
